youtube_dl/extractor/hrti.py

```diff
@@ -0,0 +1,124 @@
+            "device_model_string_id": "chrome 42.0.2311.135",
+            "application_version": HRTiIE.APP_VERSION
+        })
+
+        req = sanitized_Request(api_url, data=app_data)
+        req.get_method = lambda: 'PUT'
+
+        resources = self._download_json(req, None, note='Getting API endpoint and session information', errnote='Unable to get endpoint and session information', headers={'Content-type': 'application/json'})
+
+        self.session_id = resources['session_id']
+        self.search_url = resources['modules']['vod_catalog']['resources']['search']['uri'].format(language=HRTiIE.APP_LANGUAGE, application_id=HRTiIE.APP_PUBLICATION_ID)
+        self.login_url = resources['modules']['user']['resources']['login']['uri'].format(session_id=self.session_id) + '/format/json'
+        self.logout_url = resources['modules']['user']['resources']['logout']['uri']
+
+    def _login(self):
+        '''Performs a login to the webservice'''
+        (username, password) = self._get_login_info()
+
+        if username is None or password is None:
+            self.raise_login_required()
+
+        auth_data = json.dumps({
+            "username": username,
+            "password": password,
+        })
+        try:
+            auth_info = self._download_json(self.login_url, None, note="Authenticating", errnote="Unable to log in", data=auth_data)
+        except ExtractorError as ee:
+            if isinstance(ee.cause, compat_HTTPError) and ee.cause.code == 406:
+                raise ExtractorError('Unable to login, incorrect username and/or password')
+            raise
+
+        self.token = auth_info['secure_streaming_token']
+        self.access_token = auth_info['session_token']
+        self.logout_url = self.logout_url.format(session_id=self.session_id, access_token=self.access_token) + '/format/json'
+
+    def _real_initialize(self):
+        '''Performs necessary operations so that the information extractor is ready for operationB'''
+        self._initialize_api()
+        self._login()
+
+    def _logout(self):
+        '''Performs logout from the webservice'''
+        self._download_json(self.logout_url, None, note="Logout", errnote="Unable to log out")
+
+    def _real_extract(self, url):
+        '''Extract the data necessary to download the video'''
+        video_id = self._match_id(url)
+
+        metadata_url = self.search_url + '/video_id/{video_id}/format/json'.format(video_id=video_id)
+
+        metadata = self._download_json(metadata_url, video_id, note="Getting video metadata")
+        video = metadata['video'][0]
+        title_info = video.get('title', {})
+        title = title_info.get('title_long')
+        description = title_info.get('summary_long')
+
+        movie = video['video_assets']['movie'][0]
+        url = movie['url'].format(TOKEN=self.token)
+
+        formats = self._extract_m3u8_formats(url, video_id, 'mp4')
+
+        self._sort_formats(formats)
+
+        self._logout()
```

Comment:
Is that really required? If not just remove. If it's required `fatal=False` should be passed to `_download_json`.